package.json

```diff
@@ -5,6 +5,7 @@
   "main": "./dist/main.umd.js",
   "module": "./dist/main.es.js",
   "types": "./dist/main.d.ts",
+  "type": "module",
```

Comment:
This has been a fun deep-dive to read into. I had no idea exports were sorted. Thanks for thinking about this!

> we'll want to put the require condition above the import condition, to ensure conditions are most-to-least specific. 

Reading https://nodejs.org/api/packages.html#conditional-exports , it seems like the order of `import` and `require` is not important because they are entirely mutually exclusive. Am I missing a case where it would be ambiguous which export to use? Though, I like the idea of switching to `require -> default` (seems like a good change)

This dual-packaging example from nodejs matches our use case: https://github.com/nodejs/package-examples?tab=readme-ov-file#approach-2-isolate-state

They have the following `package.json` example:
```json
{
  "type": "module",
  "exports": {
    "import": "./index.mjs",
    "require": "./index.cjs"
  }
}
```

> We could also consider dropping the CJS export and UMD build to make things easier on ourselves

I like the idea. The dual-packaging docs mention that the global state of the cjs module is not shared with the es module -- which would interact poorly with the ts-sdks extensive use of globals. (though, admittedly, I think it is unlikely that people are exporting cjs libraries using the ts-sdk)


-- 

Deferring to the sdk team because they know more about consumption of this package.

Reply:
I've spent the last 20 minutes researching cjs vs esm in 2024 and I can't find any solid evidence that cjs is actually deprecated / dying (despite my strong bias towards esm). Multiple github issues threads on the nodejs forms indicate strongly that cjs isn't going anywhere and is still a first-class packaging format. 

So to revisit the original comment:
> We could also consider dropping the CJS export and UMD build to make things easier on ourselves

Are we actually running into difficulty with the current CJS & UMD setup? Is it worth dropping and potentially having to re-add if some customer needs to use legacy code?

I suspect the answer to both of these is a slight no. Happy to be wrong if people feel strongly.

---

I have swapped `import -> require` to `require -> default` in 2fe46c8 . I have imported this into the app and verified that everything works.

At this point, if it is okay with everyone involved, I would like to move forward with this PR mostly as-is (unless there are potentially backwards-incompatible issues).

Reply:
Agreed, I don't think any difficulty we currently face is worth dropping CJS for now.